datasource/ethereum/src/block_stream.rs

```diff
@@ -1090,28 +1099,21 @@ where
                         Ok(Async::NotReady) => {
                             // Nothing to change or yield yet.
                             state = BlockStreamState::Reconciliation(next_blocks_future);
-                            break Async::NotReady;
+                            break Ok(Async::NotReady);
                         }
 
                         Err(e) => {
                             self.consecutive_err_count += 1;
 
-                            // If too many errors without progress, give up
-                            if self.consecutive_err_count >= 100 {
-                                return Err(e);
-                            }
-
-                            warn!(
-                                self.ctx.logger,
-                                "Trying again after error in block stream reconcile: {}", e
-                            );
-
-                            // Try again by restarting reconciliation
-                            let next_blocks_future = self.ctx.next_blocks();
-                            state = BlockStreamState::Reconciliation(next_blocks_future);
-
-                            // Poll the next_blocks() future
-                            continue;
+                            // Pause before trying again
+                            let secs = (5 * self.consecutive_err_count).min(120) as u64;
+                            let instant = Instant::now() + Duration::from_secs(secs);
+                            state = BlockStreamState::Paused(Box::new(
+                                Delay::new(instant).map_err(|err| {
+                                    format_err!("Paused state's delay future failed = {}", err)
```

Comment:
Our convention is for errors to either be combined with `...: {}`, not `... = {}`, or when logged, to be added using an `error` key/value pair in the log record.